common/static/common/js/discussion/discussion_router.js

```diff
@@ -102,11 +108,43 @@
             };
 
             DiscussionRouter.prototype.showThread = function(forum_name, thread_id) {
+                var self, callback;
+                self = this;
                 this.thread = this.discussion.get(thread_id);
+
+                if (!this.thread){
```

Comment:
@e-kolpakov I remember this would happen when users got an email pointing to a specific thread. If the thread wasn't on the first page of results when loading the discussion, it would break. We had this happen a few times-- we'd apply the patches, rebase, this fix would go away, the problem would happen again, and we'd put it back.